### ISSUE TITLE
When accessing a second app in a webview for a token exchange flow, how to pass the ID token and device secret?

### ISSUE THREAD

**Issue by MeghaR2408** [okta/okta-mobile-kotlin#230]
### Describe the feature request?

Hello, I used this sample code to implement the Token Exchange flow, and it's working fine. I can use SSO to log into one app (a mobile app) and another (a web app). Both of my applications are native ones on Okta. The token exchange flow isn't working when I open a second app, a web app in Web View, and I'm not already logged in. In order for me to be logged into the second app when I access app 1 from app 2, how can I pass ID token and device secret?

### New or Affected Resource(s)

Token exchange Flow

**Comment by mikenachbaur-okta** [comment]
Hi @MeghaR2408, my apologies for the late reply.

The okta-mobile-kotlin and okta-mobile-swift SDKs do not supply any built-in patterns or mechanisms for sharing the device secret with other applications, since this is very dependent on the individual application itself.

**Comment by FeiChen-okta** [comment]
Closing this out — as noted above, the SDK intentionally does not provide a built-in mechanism for sharing the ID token/device secret between separate apps in a WebView, since this is very app-specific. If you're still blocked on a supported pattern, please open a new issue with more detail on your specific integration and we're happy to take another look.
